code/modules/power/supermatter/supermatter.dm

```diff
@@ -382,7 +393,10 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 	else
 		// Pass all the gas related code an empty gas container
 		removed = new()
-
+	overlays.Cut()
+	if(psyCoeff > 0)
+		psyOverlay.alpha = psyCoeff * 255
+		overlays += psyOverlay
```

Comment:
Instead of removing every overlay and adding this, store in a boolean variable whether our overlay is added or not, and add it if not, like I asked before.
Other things might want to overlay and then this will have to be refactored.

Reply:
Ohhh I see, will do.

Reply:
Does this do? Not totally sure I understood that